library/cgi/escapeURIComponent_spec.rb

```diff
@@ -0,0 +1,41 @@
+require_relative '../../spec_helper'
+require 'cgi'
+
+describe "CGI.escapeURIComponent" do
+  ruby_version_is "3.2" do
+    it "escapes whitespace" do
+      str = "&<>\" \xE3\x82\x86\xE3\x82\x93\xE3\x82\x86\xE3\x82\x93"
+      CGI.escapeURIComponent(str).should == '%26%3C%3E%22%20%E3%82%86%E3%82%93%E3%82%86%E3%82%93'
+    end
+
+    it "does not escape with unreserved characters" do
+      str = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-._~"
+      CGI.escapeURIComponent(str).should == "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-._~"
+    end
+
+    it "with invalid bytesequence" do
+      str = "\xC0\<\<".force_encoding("UTF-8")
+      CGI.escapeURIComponent(str).should == "%C0%3C%3C"
+    end
+
+    it "nil raises a TypeError" do
+      -> { CGI.escapeURIComponent(nil) }.should raise_error(TypeError)
```

Comment:
**suggestion:** usually it makes sense to add exception message when it's stable among supported Ruby versions.

Reply:
Here I'm a bit uncertain too... I now added an additional assertion for the error message, is this what you meant? If not, let me know so I can fix it.

Reply:
Thanks for that hint. I managed to miss that one in `CONTRIBUTING.md`, apologies. Will fix it.